File: jdacommandinterface-core/src/main/java/io/github/romangraef/jdacommandinterface/core/CommandDescription.java
package io.github.romangraef.jdacommandinterface.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface CommandDescription {
    String name();

    String[] triggers();

    String longDescription();

    String description();

    boolean hidden() default false;

    String[] usage();
}

File: jdacommandinterface-core/src/main/java/io/github/romangraef/jdacommandinterface/core/CommandDescription.kt
package io.github.romangraef.jdacommandinterface.core

@Retention(AnnotationRetention.RUNTIME)
annotation class CommandDescription(
        val name: String,
        val triggers: Array<String> = [],
        val longDescription: String = "",
        val description: String = "",
        val hidden: Boolean = false,
        val usage: Array<String> = []
)